### ISSUE TITLE
dracut-network and rd.neednet is not configuring network on Fedora

### ISSUE THREAD

**Issue by jbaublitz** [dracutdevs/dracut#1003]
**Describe the bug**
After specifying `rd.neednet=1` and `ip=ens3:dhcp` in the kernel command line, the network is not coming up.

**Distribution used**
Fedora 33

**Dracut version**
050-64.git20200529.fc33

**Init system**
systemd

**To Reproduce**
This one is going to be a little trickier because the code that I'm developing for use with the dracut module has not been released to Fedora yet and the setup is very complicated. I think a minimal example that would give a good idea of if this is reproducible on your end is using the `network` and `network-manager` modules to set up a filesystem that requires a network for setup but is not a network filesystem (such as an encrypted volume bound to a Tang server using Clevis).

**Expected behavior**
The network should be configured via dhcp on ens3 by the time clevis is used to unlock the encrypted device.

**Additional context**
I tried force loading the ethernet module with `rd.driver.pre=` and this merely made the interface available earlier. It did not force the configuration of a connection on the network.

Let me know if you need any additional details.

**Comment by jbaublitz** [comment]
I've done some additional debugging and if I manually set up the network in the emergency console, the system does boot. The interface appears to be down in the emergency console so:
```
ip set link ens3 up
ip addr add 192.168.122.180/24 dev ens3
```
appears to resolve the network connectivity issue and then using Stratis to drive unlocking via Clevis succeeds and the system boots.

**Comment by cgwalters** [comment]
For Fedora CoreOS BTW we commonly do networking in the initramfs for https://github.com/coreos/ignition/ and we also did a lot of work to enable Tang with Ignition, and support for that is covered in our test suite today.  We also use NetworkManager in the initramfs.

**Comment by jbaublitz** [comment]
I got a little more information, but hadn't gotten around to posting it here quite yet. It appears that without the `OnFailure=emergency.target` in my generated unit file, the network does indeed get configured, but there is no way to impose ordering for my service that requires the network and the network coming up itself. I think I tracked this down to the `network` module not including `NetworkManager-wait-online.service` which is wanted by `network-online.target` according to my research. Is it worth including that and its dependencies by default in the `network` module?
